poweriq-worker/src/main/java/com/vmware/flowgate/poweriqworker/jobs/PowerIQService.java

```diff
@@ -702,10 +733,25 @@ public Asset fillLocation(Sensor sensor, Map<Long, Rack> racksMap, Map<Long, Row
             assetToUpdate.setCountry(asset.getCountry());
             assetToUpdate.setRegion(asset.getRegion());
             assetToUpdate.setSerialnumber(asset.getSerialnumber());
-            HashMap<String, String> justificationfields = assetToUpdate.getJustificationfields();
-            justificationfields.put(FlowgateConstant.PDU_ID_FROM_POWERIQ, asset.getJustificationfields().get(FlowgateConstant.PDU_ID_FROM_POWERIQ));
-            justificationfields.put(FlowgateConstant.SENSOR_ID_FROM_POWERIQ, asset.getJustificationfields().get(FlowgateConstant.SENSOR_ID_FROM_POWERIQ));
-            assetToUpdate.setJustificationfields(justificationfields);
+            HashMap<String, String> oldjustificationfields = assetToUpdate.getJustificationfields();
+            Map<String,String> oldPduInfoMap = null;
+            HashMap<String, String> newSensorjustificationfields = asset.getJustificationfields();
+            Map<String,String> newPduInfoMap = null;
+            try {
+               oldPduInfoMap = getPduInfoMap(oldjustificationfields.get(FlowgateConstant.PDU));
+               newPduInfoMap = getPduInfoMap(newSensorjustificationfields.get(FlowgateConstant.PDU));
+            } catch (IOException e) {
+               logger.error("Format pdu info map error", e);
+            }
+            oldPduInfoMap.put(FlowgateConstant.PDU_ID_FROM_POWERIQ, newPduInfoMap.get(FlowgateConstant.PDU_ID_FROM_POWERIQ));
```

Comment:
It's a sensor , so I'm confused why we use PDU_ID_FROM_POERIQ ?  I guess only PDU type will use this field.